### DISCUSSION TITLE
Configuration as code section missing

### DISCUSSION THREAD

**Discussion by djurkovicp** [codemagic-ci-cd/codemagic-docs#1429] [Q&A] [answered]
Hi,
I expect this to be something simple, at least I hope so - I am missing _Configuration as code_ section when I'm in the _Workflow Editor_.
I'd like to fill in some basic configuration in the workflow editor and then fine tune it. This is not anymore on the right side and not only this, everything that was at one point below the _Badge_ config on the right is now moved to different location.

The old layout I'm referencing is present in the [guides](https://blog.codemagic.io/codemagic-yaml-cheatsheet/) as well. 

_Repository setting_ and _Scheduled build_ config, that was on the right as well, is quite visible, but despite my efforts I failed to find the location of the _Configuration as code_ in this (new) layout.
I am only able to switch to YAML configuration, but once there I get a blank yaml and I don't see an option there to fill in from the _Workflow editor_.

**Comment by icarusdust** [comment]
Hi @djurkovicp, sorry to hear that you faced difficulties finding it. Please note that we have removed that option, it will give you a blank page because your yaml file is not present in the project. In anyway, even with the old option you would have to configure codemagic.yaml manually in your project as Codemagic has no write access to your repo. If you can share on what platform you are building apps, I can send you a sample codemagic.yaml where all the steps are clearly explained.

**Reply by hellyab** [reply]
It would have been nice to keep the generate yaml option because it allows for an easy transition from the workflow builder to the yaml config. Now I am also left with the daunting task of rewriting an old configuration from a year ago.

**Comment by djurkovicp** [comment, marked as answer]
Apparently it's fixed today. Now when you go to _Switch to YAML configuration_ you get your _Workflow Editor_ configuration in the YAML. No more empty yaml file.

**Reply by ricardolb** [reply]
I couldn't get this to work, did you have to do anything else?

**Reply by pdjurkovic** [reply]
I wish I could help you, but I don't see anything meaningful in my actions that made it work.
I see now it's back to blank YAMLs again if I try in my app... so back to having this as an issue.

**Reply by icarusdust** [reply]
@djurkovicp I am glad to hear that you could find your yaml but I must say that it is not automatically written into yaml and again automatically placed into your repo. It is technically impossible at first because Codemagic does not have write access to you repo. So, you created codemagic.yaml file manually in the root dir of your project and only then it was detected by Codemagic. Just for more context here, we removed it because it was using an older way of configuring environment variables. 
@pdjurkovic It is blank because you do not have codemagic.yaml configured in your project. I can help you with a ready yaml file to configure it if you can let me know what platform you are building your apps with

**Reply by pdjurkovic** [reply]
Nobody with the problem here is talking about Codemagic having to automatically include the yaml file into the repo, it's about having a half-baked starting point generated from the GUI/Workflow editor, that we could continue to build on and include it into the repo.

**Reply by icarusdust** [reply]
I totally understand your concern but please note that as it was using an older way to configure a yaml file, we decided to remove it.

**Reply by f-person** [reply]
@icarusdust that's quite sad. I want to switch to YAML, but I'd love to avoid dealing with distribution-related stuff manually :). 
The nice thing about "Download configuration" was the ability to have a good starting point.
